File: src/core/jupiter/core/domain/concept/time_plans/time_plan_source.py
"""The source of a time plan."""

from jupiter.core.framework.value import EnumValue, enum_value


@enum_value
class TimePlanSource(EnumValue):
    """The source of a time plan."""

    USER = "user"
    GENERATED = "generated"

    @property
    def allow_user_changes(self) -> bool:
        """Whether the user can change the time plan."""
        # Keep synced with ts:time-plan-source.ts
        return self == TimePlanSource.USER

File: src/webui/app/logic/domain/time-plan-source.ts
import { TimePlanSource } from "@jupiter/webapi-client";

export function timePlanSourceName(source: TimePlanSource): string {
  switch (source) {
    case TimePlanSource.USER:
      return "User";
    case TimePlanSource.GENERATED:
      return "Recurring";
  }
}

export function allowUserChanges(source: TimePlanSource): boolean {
  // Keep synced with python:time-plan-source.py
  return source === TimePlanSource.USER;
}
